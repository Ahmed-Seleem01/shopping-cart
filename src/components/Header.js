import icon from "../images/ingredients.png";
import cart from "../images/shopping-cart.png";
import "../styles/header.scss";
import { Link } from "react-router-dom";
import { useEffect, useState } from "react";

import Cart from "./Cart";

const Header = ({ headerCart }) => {
  const [quantity=0] = headerCart;
  const [cartItems, setCartItems] = useState(0);
  const [showCart, setShowCart] = useState("cart-container");
  console.log(cartItems);

  useEffect(() => {
    setCartItems(Number(cartItems) + 1);
  }, [headerCart]);

  const handleToggle = ()=>{
    setShowCart("cart-container toggle-cart")
  }
  const hide = ()=>{
    setShowCart("cart-container")
  }
  return (
    <nav className="header">
      <div className="header-left">
        <img src={icon} alt="site icon" />
        <Link to="/">
          <h1>Ingrediento</h1>
        </Link>
      </div>
      <div className="header-right">
        <div className="global-links">
          <Link to="/">
            <h2>Homepage</h2>
          </Link>
          <Link to="/shopping">
            <h2>Shopping</h2>
          </Link>
        </div>
        <div className="cart" onClick={handleToggle}>
          <sup>{cartItems -1}</sup>
          <img src={cart} alt="cart icon" />
        </div>
      </div>
      <Cart hide={hide} showCart= {showCart} headerCart={headerCart}/>
    </nav>
  );
};

export default Header;
