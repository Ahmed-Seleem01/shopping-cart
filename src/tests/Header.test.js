import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "../components/Header";

describe("test the whole component", () => {
  it("check the snapshot", ()=>{
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  })
});

describe("test the header", () => {
  it("check the site icon", () => {
    render(<Header />);
    const icon = screen.getByAltText("site icon");
    expect(icon.alt).toBe("site icon");
  });

  it("check the site title", () => {
    render(<Header />);
    const title = screen.getByRole("heading");
    expect(title.textContent).toBe("Ingrediento");
  });

  it("check the cart icon", () => {
    render(<Header />);
    const icon = screen.getByAltText("cart icon");
    expect(icon.alt).toBe("cart icon");
  });
});
