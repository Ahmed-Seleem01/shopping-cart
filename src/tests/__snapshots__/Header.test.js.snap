// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the whole component check the snapshot 1`] = `
<div>
  <div
    class="header"
  >
    <div>
      <img
        alt="site icon"
        src="ingredients.png"
      />
      <h1>
        Ingrediento
      </h1>
    </div>
    <img
      alt="cart icon"
      src="shopping-cart.png"
    />
  </div>
</div>
`;
